### ISSUE TITLE
Support for PostgreSQL's Money type

### ISSUE THREAD

**Issue by ivan-m** [lpsmith/postgresql-simple#205]
None of the instances of `FromField` seem to support the [`money`](https://www.postgresql.org/docs/current/static/datatype-money.html) type; is this deliberate or an oversight?

It seems as if it should be able to convert to/from `Double`.

**Comment by reiddraper** [comment]
I might imagine something from `Data.Fixed` or `Data.Scientific` might be more appropriate. I'd avoid using `Double` for currency calculations.

**Comment by lpsmith** [comment]
I second Reid.  Also,  I prefer that `FromField` instances provided with `postgresql-simple` reflect the type(s) they support with as full fidelity as possible.  IIRC, the money type is also tagged with a currency code.   Adding support for `Data.Fixed` and/or `Data.Scientific` might be acceptable, but really you probably need a type specific to handling `money`.

**Comment by lpsmith** [comment]
Also keep in mind there is no type checking or anything on `ToField` instances in postgresql-simple.

**Comment by ivan-m** [comment]
OK, so I'm trying to use `Fixed`, but I can't seem to work out how to specify the proper precision on the Postgre side, even for `NUMERIC(precision, scale)` (specifically for the `scale` component to ensure it maps to the correct `HasResolution` instance).

**Comment by lpsmith** [comment]
Yeah,  I looked at that issue once and run into trouble myself.  I don't recall the specifics,  but I recall I wasn't overly pleased with postgres's handling of the numeric type.

**Comment by Superpat** [comment]
Opaleye has a similar discussion going on https://github.com/tomjaguarpaw/haskell-opaleye/issues/230

I personally use Numeric(15,6) to represent money, from what I read Data.Decimal seems like a good choice for a haskell datatype but maybe Fixed could work as well.
